kserve/odh-overlays/controller/inferenceservice-config-patch.yaml

```diff
@@ -0,0 +1,60 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: inferenceservice-config
+  namespace: kserve
+data:
+  explainers: |-
+    {
+        "alibi": {
+            "image" : "$(kserve-alibi-explainer)",
+            "defaultImageVersion": "$(kserve-explainer-version)"
+        },
+        "art": {
+            "image" : "$(kserve-art-explainer)",
+            "defaultImageVersion": "$(kserve-explainer-version)"
+        }
+    }
+  storageInitializer: |-
+    {
+        "image" : "$(kserve-storage-initializer)",
+        "memoryRequest": "100Mi",
+        "memoryLimit": "1Gi",
+        "cpuRequest": "100m",
+        "cpuLimit": "1",
+        "storageSpecSecretName": "storage-config",
+        "enableDirectPvcVolumeMount": false
+    }
+  logger: |-
+    {
+        "image" : "$(kserve-agent)",
+        "memoryRequest": "100Mi",
+        "memoryLimit": "1Gi",
+        "cpuRequest": "100m",
+        "cpuLimit": "1",
+        "defaultUrl": "http://default-broker"
+    }
+  batcher: |-
+    {
+        "image" : "$(kserve-agent)",
+        "memoryRequest": "1Gi",
+        "memoryLimit": "1Gi",
+        "cpuRequest": "1",
+        "cpuLimit": "1"
+    }
+  agent: |-
+    {
+        "image" : "$(kserve-agent)",
+        "memoryRequest": "100Mi",
+        "memoryLimit": "1Gi",
+        "cpuRequest": "100m",
+        "cpuLimit": "1"
+    }
+  router: |-
+    {
+        "image" : "$(kserve-router)",
+        "memoryRequest": "100Mi",
+        "memoryLimit": "1Gi",
+        "cpuRequest": "100m",
+        "cpuLimit": "1"
+    }
```

Comment:
This will always overwrite the original file because it is 1 json object. So I think it would be better to update original file for `the image part`
For example, if router param in the original file changed like following 
~~~
  router: |-
    {
       "test":"test",  <---
        "image" : "$(kserve-router)",
        "memoryRequest": "100Mi",
        "memoryLimit": "1Gi",
        "cpuRequest": "100m",
        "cpuLimit": "1"
    }
~~~
It will not be applied because this file overwrite the one.

Reply:
The idea was to keep the `kserve` resources as it to be able to update them via the [script](https://github.com/opendatahub-io/odh-manifests/pull/838/files#diff-80e7482e52299d84ae57e4680cf757973a1049eb83e088fa2c96d49f01b27e43). If we change the original file, it gets a bit harder to update. If we want to keep that approach, we need to replace the whole block here, as I don't think kustomize supports replacing sub-parts of a inline yaml. 

I'm fine with both ways. WDYT?

Reply:
Perhaps a custom transformer: https://kubectl.docs.kubernetes.io/guides/extending_kustomize/. But I'm not sure if we want to use custom transformers.

If not, I don't see other way of doing it. So, I agree this is quite fine for now.